src/main/webapp/app/shared/util/core-evidence-submission/type-specific-resolvers.ts

```diff
@@ -90,6 +91,26 @@ function handleGeneSummary({ gene, evidenceData }: Pick<GetEvidenceArgs, 'gene'>
   evidenceData.data.lastEdit = validateTimeFormat(gene.summary_review?.updateTime);
 }
 
+function handleMutationNameChange({
+  evidenceData,
+  gene,
+  mutation,
+}: Pick<GetEvidenceArgs, 'gene' | 'mutation'> & { evidenceData: InitializeEvidenceDataRtn }) {
+  evidenceData.data.alterations = getAlterations(gene.name, mutation);
+}
+
+function handleTumorNameChange({ evidenceData, tumor }: Pick<GetEvidenceArgs, 'tumor'> & { evidenceData: InitializeEvidenceDataRtn }) {
+  evidenceData.data.cancerTypes = tumor.cancerTypes as TumorType[];
+  evidenceData.data.excludedCancerTypes = tumor.excludedCancerTypes as TumorType[];
+}
+
+function handleTreatmentNameChange({
+  evidenceData,
+  treatment,
+}: Pick<GetEvidenceArgs, 'treatment'> & { evidenceData: InitializeEvidenceDataRtn }) {
+  evidenceData.data.excludedCancerTypes = treatment.excludedRCTs as TumorType[];
+}
+
```

Comment:
Talking about pattern matching, do you want to merge these three?

Reply:
The reason why didn't do that here is because I have dedicated functions for every case in the switch statement so I thought I would be less confusing if I wrote it this way.